dlib/CMakeLists.txt

```diff
@@ -550,31 +564,28 @@ if (NOT TARGET dlib)
 
 
       # Tell CMake to build dlib via add_library()/cuda_add_library()
-      if (DLIB_USE_CUDA)
-         cuda_add_library(dlib STATIC ${source_files} )
+      if(BUILD_SHARED_LIBS AND NOT DLIB_IN_PROJECT_BUILD)
+        if (DLIB_USE_CUDA)
+            cuda_add_library(dlib SHARED ${source_files} )
+        else()
+            add_library(dlib SHARED ${source_files} )
```

Comment:
If `BUILD_SHARED_LIBS=ON` then `add_library` will create shared library, there is no need to set `SHARED` explicitly.